ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -419,6 +426,109 @@ func SecretToGateways(namespace, name, key string) ([]string, bool) {
 	return gwNsNameList, found
 }
 
+func PodToGateway(namespace, name, key string) ([]string, bool) {
+	podNsName := namespace + "/" + name
+	pod, err := utils.GetInformers().PodInformer.Lister().Pods(namespace).Get(name)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			utils.AviLog.Infof("key: %s, got error while getting pod: %v", key, err)
+			return []string{}, false
+		}
+		utils.AviLog.Infof("key: %s, msg: Pod not found, deleting mappings", key)
+		servicesList := akogatewayapiobjects.GatewayApiLister().GetPodsToService(podNsName)
+		gatewayList := []string{}
+		for _, svcNsName := range servicesList {
+			found, gwNsNameList := akogatewayapiobjects.GatewayApiLister().GetServiceToGateway(svcNsName)
+			if found {
+				for _, gwNsName := range gwNsNameList {
+					if !utils.HasElem(gatewayList, gwNsName) {
+						gatewayList = append(gatewayList, gwNsName)
+					}
+				}
+
+			}
+		}
+		return gatewayList, true
+	}
+	ann := pod.GetAnnotations()
+	var annotations []lib.NPLAnnotation
+	if err := json.Unmarshal([]byte(ann[lib.NPLPodAnnotation]), &annotations); err != nil {
+		utils.AviLog.Infof("key: %s, got error while unmarshaling NPL annotations: %v", key, err)
+	}
+	objects.SharedNPLLister().Save(podNsName, annotations)
+
+	servicesList, _ := lib.GetServicesForPod(pod)
+	oldServicesList := akogatewayapiobjects.GatewayApiLister().GetPodsToService(podNsName)
+	for _, svc := range oldServicesList {
```

Comment:
One basic question for my understanding why services from oldserviceslist are being added to servicesList ?

Reply:
GetServicesForPod retrieves the current llst of services associated with the pod. this only gives the list of added or updated resources. resources that previously had the mapping or those in the deleted case can only be retrieved from oldServicesList.

oldServicesList helps us map gateways that need to have the mappings removed